### PR TITLE
Add support for public, overloaded, bincompat copy

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by dwijnand** [comment]
Review by @eed3si9n and @Duhemm.

@Duhemm I implemented this so we don't have to change all the sbt modules codebases to avoid copy (as well as for the plugins and users of sbt 1.0).

File: notes/0.2.6.markdown [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        case _ => false
+      }
+      override def hashCode: Int = {
+        37 * (37 * (17 + x.##) + y.##)
+      }
+      override def toString: String = {
+        "Foo(" + x + ", " + y + ")"
+      }
+      def copy(): Foo = {
+        new Foo(0, 0)
```
Comment (eed3si9n) [inline notes/0.2.6.markdown]: How is that  a copy?

File: notes/0.2.6.markdown [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      }
+      def copy(x: Int): Foo = {
+        new Foo(x, 0)
```
Comment (eed3si9n) [inline notes/0.2.6.markdown]: So `Foo(1, 2).copy(x = 7)` will generate `Foo(7, 0)`?

**Comment by dwijnand** [comment]
Good catch, thanks.

**Pull request merged**